### DISCUSSION TITLE
WiFi not getting DHCP lease on ESP32 core v2.0.0 or v2.0.1

### DISCUSSION THREAD

**Discussion by gomegs** [khoih-prog/ESPAsync_WiFiManager#73] [Q&A]
Hi,

Hope someone can help.

I have code using ESPAsync_WiFiManager and ESP32 core 1.0.6 that works and the ESP32 get's a DHCP lease from my router as expected but after moving to ESP32 core v2.0.0 or 2.0.1 WiFi.localIP() returns 255.255.255.255 after WiFi is connected.

What gives?

Using Arduino 1.8.13 with Arduino core for the ESP32
https://github.com/espressif/arduino-esp32

**Serial output on v1.0.6**
ESP32 core v1_0_6
ESP32 Chip model = ESP32-D0WDQ6 Rev 0
Starting Async_ConfigOnDoubleReset_minimal on ESP32_DEV
ESPAsync_WiFiManager v1.9.4
[D][WiFiGeneric.cpp:374] _eventCallback(): Event: 0 - WIFI_READY
[D][WiFiGeneric.cpp:374] _eventCallback(): Event: 2 - STA_START
WiFi connecting, in... [D][WiFiGeneric.cpp:374] _eventCallback(): Event: 4 - STA_CONNECTED
[D][WiFiGeneric.cpp:374] _eventCallback(): Event: 7 - STA_GOT_IP
[D][WiFiGeneric.cpp:419] _eventCallback(): STA IP: 172.26.1.214, MASK: 255.255.255.0, GW: 172.26.1.3
2.80 secs, Connection result is 3
Local IP: 172.26.1.214

**Serial output on v2.0.1**
ESP32 core v2_0_1
ESP32 Chip model = ESP32-D0WDQ6 Rev 0
Starting Async_ConfigOnDoubleReset_minimal on ESP32_DEV
ESPAsync_WiFiManager v1.9.4
[   462][D][WiFiGeneric.cpp:831] _eventCallback(): Arduino Event: 0 - WIFI_READY
[   560][V][WiFiGeneric.cpp:96] set_esp_interface_ip(): Configuring Station static IP: 255.255.255.255, MASK: 255.255.255.255, GW: 255.255.255.255
[   560][V][WiFiGeneric.cpp:272] _arduino_event_cb(): STA Started
[   568][D][WiFiGeneric.cpp:831] _eventCallback(): Arduino Event: 2 - STA_START
WiFi connecting, in... [  2680][V][WiFiGeneric.cpp:284] _arduino_event_cb(): STA Connected: SSID: SomeSSID, BSSID: ce:2d:e0:59:4b:5f, Channel: 1, Auth: WPA2_PSK
[  2681][D][WiFiGeneric.cpp:831] _eventCallback(): Arduino Event: 4 - STA_CONNECTED
[  2682][V][WiFiGeneric.cpp:294] _arduino_event_cb(): STA Got New IP:255.255.255.255
[  2696][D][WiFiGeneric.cpp:831] _eventCallback(): Arduino Event: 7 - STA_GOT_IP
[  2703][D][WiFiGeneric.cpp:880] _eventCallback(): STA IP: 255.255.255.255, MASK: 255.255.255.255, GW: 255.255.255.255
2.20 secs, Connection result is 3
Local IP: 255.255.255.255

Anybody else having this issue?

Thank you and kind regards.

**Comment by khoih-prog** [comment]
Hi,

Thanks for your interest and report of the issue.

Due to recent breaking major change from ESP32 core v1.0.6 to core v2.0.0+, the [**Async_ConfigOnDRD_ESP32_minimal**](https://github.com/khoih-prog/ESPAsync_WiFiManager/tree/master/examples/Async_ConfigOnDRD_ESP32_minimal) can't run normally as in the past with v1.0.6

You also have to understand that these `so-called minimal` examples, such as the above mentioned example, was written just to demonstrate minimal basic features of the library, and please don't expect much from that kind of `minimal` example.

I also won't spend time to investigate and fix those `minimal` examples and will even **delete all of them in the near future**, because t**hey create more issues than benefits to users**.

I suggest to move to use better examples, such as [**Async_ConfigOnDoubleReset**](https://github.com/khoih-prog/ESPAsync_WiFiManager/tree/master/examples/Async_ConfigOnDoubleReset), which will r**un normally with core 2.0.0+.**

Good Luck,
